cpp/ql/lib/semmle/code/cpp/models/interfaces/PartialFlow.qll

```diff
@@ -0,0 +1,15 @@
+import semmle.code.cpp.Function
+import FunctionInputsAndOutputs
+import semmle.code.cpp.models.Models
+
+/**
+ * A function that may (but not always) updates (part of) a `FunctionOutput`.
+ */
+abstract class PartialFlowFunction extends Function {
```

Comment:
This is a FP since [StrDupPtr](https://github.com/github/codeql/blob/be9759fbad706c784e7fc2a8b0e8a977ec23de37/cpp/ql/test/library-tests/dataflow/dataflow-tests/TestBase.qll#L68-L68) is only in test files.